project2/screens/search.js

```diff
@@ -1,131 +1,72 @@
-        <View style={styles.row}>
-            <Text style={styles.listText}>{props.title}</Text>
-        </View>
-    </TouchableOpacity>
-)
-const renderItem = selected => obj => <Row selected={selected}{...obj.item} />
-
-export default class search extends React.Component {
-    static navigationOptions = {
-        title: "Search",
-        headerTitleStyle: { color: "#AAAAAF", }
-    };
-
-    state = {
-        searchString: "",
-        movies: "",
-        totalResults: 0,
-        totalPages: 0,
-        nextPage: 0,
-    };
-
-    getMovies = async (searchString) => {
-        const results = await searchMovies(`${searchString}`)
-        console.log(results)
-
-        if (results !== undefined) {
-            this.setState({ movies: results["result"] })
-            this.setState({ totalResults: results["totalResults"] })
-            this.setState({ totalPages: results["totalPages"] })
-
-            if (results["totalPages"] > 1) {
-                this.setState({ nextPage: 2 })
-            }
-        }
-    }
-
-    componentDidUpdate(prevProps, prevState) {
-        if (this.state.searchString !== prevState.searchString) {
-            this.getMovies(this.state.searchString)
+import SearchListItem from "./Search/ListItem";
+import s from "../Styles";
+
+const Search = () => {
+  const [searchString, setSearchString] = React.useState("");
+  const [movies, setMovies] = React.useState([]);
+  const [totalResults, setTotalResults] = React.useState(0);
+  const [totalPages, setTotalPages] = React.useState(0);
+  const [nextPage, setNextPage] = React.useState(0);
+
+  React.useEffect(() => {
+    const getMovies = async () => {
+      const { result, totalResults, totalPages } = await searchMovies(
+        searchString
+      );
+
+      if (result) {
+        setMovies(result);
+        setTotalResults(totalResults);
+        setTotalPages(totalPages);
+
+        if (totalPages > 1) {
+          setNextPage(2);
         }
-    }
-
-    updateSearch = searchString => {
-        this.setState({
-            searchString: searchString,
-            movies: "",
-            totalResults: 0,
-            totalPages: 0,
-            nextPage: 0
-        })
-    }
-
-    movieSelected = (id) => {
-        this.props.navigation.navigate("Movie", { "MovieID": id })
-    }
-
-    displayMore = async () => {
-        if (this.state.totalPages === 1) return
-        if (this.state.nextPage === this.state.totalPages) return
-
-        const results = await getMoviePage(this.state.searchString, this.state.nextPage)
-
-        if (results !== undefined) {
-            this.setState(prevState => ({ movies: [...prevState.movies, ...results] }));
+      }
+    };
 
-            if (this.state.nextPage < this.state.totalPages) {
-                this.setState({ nextPage: this.state.nextPage++ })
-            }
-        }
+    if (searchString !== "") {
+      getMovies();
     }
+  }, [searchString]);
```

Comment:
In the array of dependencies, I'm only defining the searchString state. This means the effect will fire once when the component mounts (which won't actually run anything, as it should onyl run when it's not an empty string), and then once every time the searchString changes.

I about it being preferable to use a `useState` for every single value in the state -- this is an example of why. React will compare `searchString` with the previous value. As it's a string, it's directly comparable, so this will all work fine. As a comparisong, if I have one `useState` at the top of the component, like:

```
 const [state, setState] = React.useState({ searchString: "", movies: [], totalResults: 0, totalPages: 0, nextPage: null})
```

That would be an object, and React wouldn't be able to compare the previous to the current value properly; generally what happens then is everything goes into an infinite loop, as it guesses that things have updated, so it rerenders, then runs the effect, which guesses there's been an update, and so on and so forth.